File: app/src/main/java/com/studyasist/data/local/entity/Assessment.kt
package com.studyasist.data.local.entity

import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.Index
import androidx.room.PrimaryKey

@Entity(
    tableName = "assessments",
    foreignKeys = [
        ForeignKey(
            entity = Goal::class,
            parentColumns = ["id"],
            childColumns = ["goalId"],
            onDelete = ForeignKey.SET_NULL
        )
    ],
    indices = [Index(value = ["goalId"]), Index(value = ["subject"]), Index(value = ["chapter"])]
)
data class Assessment(
    @PrimaryKey(autoGenerate = true)
    val id: Long = 0,
    val title: String,
    val goalId: Long? = null,
    val subject: String? = null,
    val chapter: String? = null,
    val totalTimeSeconds: Int,
    val randomizeQuestions: Boolean = true,
    val createdAt: Long = System.currentTimeMillis()
)

File: StudyAsist-iOS/StudyAsist/Models/Assessment.swift
//
//  Assessment.swift
//  StudyAsist
//

import Foundation
import SwiftData

@Model
final class Assessment {
    @Attribute(.unique) var id: Int64
    var title: String
    var goalId: Int64?
    var subject: String?
    var chapter: String?
    var totalTimeSeconds: Int
    var randomizeQuestions: Bool
    var createdAt: Int64
    
    init(
        id: Int64 = 0,
        title: String,
        goalId: Int64? = nil,
        subject: String? = nil,
        chapter: String? = nil,
        totalTimeSeconds: Int,
        randomizeQuestions: Bool = true,
        createdAt: Int64 = Int64(Date().timeIntervalSince1970 * 1000)
    ) {
        self.id = id
        self.title = title
        self.goalId = goalId
        self.subject = subject
        self.chapter = chapter
        self.totalTimeSeconds = totalTimeSeconds
        self.randomizeQuestions = randomizeQuestions
        self.createdAt = createdAt
    }
}
